lumigator/python/mzai/backend/backend/services/experiments.py

```diff
@@ -0,0 +1,168 @@
+import asyncio
+import csv
+import json
+from collections.abc import Callable
+from io import BytesIO, StringIO
+from pathlib import Path
+from uuid import UUID
+
+import loguru
+from fastapi import BackgroundTasks, UploadFile
+from lumigator_schemas.datasets import DatasetFormat
+from lumigator_schemas.experiments import ExperimentCreate, ExperimentResponse
+from lumigator_schemas.jobs import (
+    JobEvalCreate,
+    JobInferenceCreate,
+    JobStatus,
+)
+from s3fs import S3FileSystem
+
+from backend.services.datasets import DatasetService
+from backend.services.jobs import JobService
+from backend.settings import settings
+
+
+class ExperimentService:
+    def __init__(self, job_service: JobService, dataset_service: DatasetService):
+        self._job_service = job_service
+        self._dataset_service = dataset_service
+
+    def _add_dataset_to_db(self, job_id: UUID, request: JobInferenceCreate):
+        loguru.logger.info("Adding a new dataset entry to the database...")
+        s3 = S3FileSystem()
+
+        # Get the dataset from the S3 bucket
+        result_key = self._job_service._get_results_s3_key(job_id)
+        with s3.open(f"{settings.S3_BUCKET}/{result_key}", "r") as f:
+            results = json.loads(f.read())
+
+        dataset = {
```

Comment:
non-blocking, suggestion to make this slightly more legible by pulling out the fields for emphasis:

```python
fields = ["examples", "ground_truth", request.output_field]
dataset = {k: v for k, v in results.items() if k in fields}
```